File: 100dynamic/2466.py
class Solution:
    def countGoodStrings(self, low: int, high: int, zero: int, one: int) -> int:
        MOD = 1_000_000_007
        f = [1] + [0] * high  # f[i] 表示构造长为 i 的字符串的方案数
        for i in range(1, high + 1):
            if i >= zero: f[i] = (f[i] + f[i - zero]) % MOD
            if i >= one:  f[i] = (f[i] + f[i - one]) % MOD
        return sum(f[low:]) % MOD

File: 100dynamic/2466.rs
impl Solution {
    pub fn count_good_strings(low: i32, high: i32, zero: i32, one: i32) -> i32 {
        const MOD: i32 = 1_000_000_007;
        let mut ans = 0;
        let mut f = vec![0; (high + 1) as usize];
        f[0] = 1;
        for i in 1..=high as usize {
            if i >= zero as usize {
                f[i] = (f[i] + f[i - zero as usize]) % MOD;
            }
            if i >= one as usize {
                f[i] = (f[i] + f[i - one as usize]) % MOD;
            }
            if i >= low as usize {
                ans = (ans + f[i]) % MOD;
            }
        }
        ans
    }
}